### PR TITLE
Spring 4 update

### PR DESCRIPTION
**Pull request by biconou**
Pull request to upgrade to Spring 4. 
This is the next step to move to SpringBoot as defined in the 6.2 roadmap.

Spring target version : **4.3.4.RELEASE** (because it is a dependency of SpringBoot 1.4.2.RELEASE)

New "temporary" version of libresonic : **6.2.beta1.spring4**

All deprecated SimpleFormController must migrate using the @Controller annotation.

### CODE REVIEW

**Comment by EugeneKay** [comment]
This looks pretty good at first glance. I'll wait until tonight to merge it in, unless somebody else reports a problem first.

**Comment by biconou** [comment]
@EugeneKay Wait. This is a WIP. I am on the way to migrate all deprecated controllers (SimpleFormController). Then change Spring version to 4.

**Comment by muff1nman** [comment]
We'll want to do a rebase on this too, as some of these commits are already in develop which rebase will take care of

**Comment by biconou** [comment]
Migrating the controllers is a bit boring :-1: 
I work.

**Comment by muff1nman** [comment]
@biconou I've been converting to liquibase xml, wanna swap? haha

**Comment by biconou** [comment]
@muff1nman Sure ! I am very interested by liquibase but never had the opportunity to use it in real life.
But, at present, I must concentrate on Spring migration.

**Comment by biconou** [comment]
UserSettingsController is a mess. :-1:

File: libresonic-assembly/pom.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <parent>
         <groupId>org.libresonic.player</groupId>
         <artifactId>libresonic</artifactId>
-        <version>6.1.beta2</version>
+        <version>6.2.spring4-SNAPSHOT</version>
```
Comment (zifnab06) [inline libresonic-assembly/pom.xml]: Snapshots are not idempotent in maven, can you please pick a released version thats guaranteed to be around later?

Comment (biconou) [inline libresonic-assembly/pom.xml]: What version number would you suggest ? 
Is 6.2.beta1.spring4 ok ?

**Comment by biconou** [comment]
Migrating to Spring 4 implies a CXF migration.

File: libresonic-assembly/pom.xml (hunk 1 above)
Comment (zifnab06) [inline libresonic-assembly/pom.xml]: I was half asleep or something when I commented here - I thought this was the spring version. Please ignore :P

Comment (biconou) [inline libresonic-assembly/pom.xml]: Ok. Indeed I didn't quite understand.

**Comment by biconou** [comment]
Spring security will stay at version 3.2.9.RELEASE cause migrating to the latest version needs code migration and 3.2.9.RELEASE seems to be still compatible with Spring 4.

**Comment by biconou** [comment]
Ok, migration to Spring 4 seems to be done.

**Comment by EugeneKay** [comment]
Travis RPM failure aside, is this ready to get merged into develop?

**Comment by biconou** [comment]
I think so (if we accept that it can be unstable). I should try to change the version number, removing the hyphen to see if Travis is happy again.

**Comment by biconou** [comment]
I am preparing a new PR for SpringBoot.

**Comment by biconou** [comment]
Travis happy again ! :-)
I think this PR could be merged.

**Comment by muff1nman** [comment]
Do you think you could rebase?

**Comment by EugeneKay** [comment]
Leeeeeeeroy..... mmmmjenkins!

**Comment by fxthomas** [comment]
This was not a complete party wipe, almost everything seems to work. Great job! :)

Apparently Spring decided to drop handling POST requests in `ParametrizableViewController` though. I've submitted a PR for one I've worked on, but some others are missing as well, such as `InternetRadioController` for instance.

**Comment by biconou** [comment]
Thanks. I am not familiar with Spring MVC; your help is valuable.

**Comment by fxthomas** [comment]
I actually only started using Spring a few months back when I started contributing to Libresonic, but I can always help with testing when I have time :)

**Pull request merged**